exonum/src/sandbox/sandbox.rs

```diff
@@ -765,21 +786,19 @@ fn gen_primitive_socket_addr(idx: u8) -> SocketAddr {
 fn sandbox_with_services_uninitialized(
     services: Vec<Box<dyn Service>>,
     consensus: ConsensusConfig,
+    validators_count: u8,
 ) -> Sandbox {
-    let validators = vec![
-        gen_keypair_from_seed(&Seed::new([12; SEED_LENGTH])),
-        gen_keypair_from_seed(&Seed::new([13; SEED_LENGTH])),
-        gen_keypair_from_seed(&Seed::new([16; SEED_LENGTH])),
-        gen_keypair_from_seed(&Seed::new([19; SEED_LENGTH])),
-    ];
-    let service_keys = vec![
-        gen_keypair_from_seed(&Seed::new([20; SEED_LENGTH])),
-        gen_keypair_from_seed(&Seed::new([21; SEED_LENGTH])),
-        gen_keypair_from_seed(&Seed::new([22; SEED_LENGTH])),
-        gen_keypair_from_seed(&Seed::new([23; SEED_LENGTH])),
-    ];
-
-    let addresses: Vec<SocketAddr> = (1..5).map(gen_primitive_socket_addr).collect::<Vec<_>>();
+    let validators = (0..validators_count)
+        .map(|i| gen_keypair_from_seed(&Seed::new([i + 12; SEED_LENGTH])))
+        .collect::<Vec<_>>();
+
+    let service_keys = (0..validators_count)
+        .map(|i| gen_keypair_from_seed(&Seed::new([i + 20; SEED_LENGTH])))
+        .collect::<Vec<_>>();
```

Comment:
When we had only four validators the seed values were all different for them. However, now if we request more than eight validators then some keys may be generated using the same seeds:

* `i + 12` for `i == 8`
* `i + 20` for `i == 0`

I think that it is important to keep the seeds fixed so that tests are reproducible, but maybe we should avoid collisions. For example, with something like this:

```rust
let mut keys = (42..).map(|i| gen_keypair_from_seed(&Seed::new([i; SEED_LENGTH]));
let validators = keys.take(validators_count).collect::<Vec<_>>();
let service_keys = keys.take(validators_count).collect::<Vec<_>>();
```

Reply:
Oh, that wouldn't work as is, but you get the idea.

I'm not sure that we need it though. It's just tests.